roles/network_plugin/calico/tasks/install.yml

```diff
@@ -331,6 +352,35 @@
     - local_as is defined
     - groups['calico-rr'] | default([]) | length == 0
 
+- name: Calico | Configure node asNumber for per node peering automatically
```

Comment:
As discussed on slack, local_as is a property specific to the node and should not be defined from the peers the node is connected to.
Moreover, this task handles a case where peer asn is the same as the local asn for a node, and it is already covered by the previous task by setting the local_as accordingly.
Finally, in case of multiple peers for each node, at best (with all peers having the same asn), the BGPConfiguration will be apply multiple times for the same node, at worst (with different asn for each peer), it will lead to unwanted behaviour.